client/rpc-api/src/system/error.rs

```diff
@@ -43,6 +45,11 @@ impl From<Error> for rpc::Error {
 				message: format!("{}", e),
 				data: serde_json::to_value(h).ok(),
 			},
+			Error::MalformattedPeerArg(ref e) => rpc::Error {
+				code :rpc::ErrorCode::ServerError(BASE_ERROR + 2),
+				message: e.clone(),
+				data: Some(e.to_string().into()),
```

Comment:
`e` is just the string, so `data` would be just the same as `message`, right? I think we can just skip it then. `data` is meant to contain more detailed information about the error.
```suggestion
				data: None,
```